file-formats/rvbc/type/zif_aff_rvbc_v1.intf.abap

```diff
@@ -0,0 +1,93 @@
+"! AFF type for the object RVBC ( Review Booklets )
+INTERFACE zif_aff_rvbc_v1
+  PUBLIC.
+  "! $values { @link zif_aff_rvbc_v1.data:co_extensibility_mode }
+  "! $default { @link zif_aff_rvbc_v1.data:co_extensibility_mode.not_supported }
+  TYPES ty_extensibility_mode TYPE n LENGTH 2.
+  TYPES ty_app_id             TYPE c LENGTH 70.
+  TYPES ty_source             TYPE n LENGTH 1.
+  TYPES ty_status             TYPE n LENGTH 1.
+  TYPES ty_consistency_status TYPE n LENGTH 1.
+  TYPES ty_ina1_service_id    TYPE c LENGTH 30.
+  TYPES ty_ina1_service_name  TYPE c LENGTH 40.
+  TYPES ty_ina1_service_type  TYPE n LENGTH 2.
+
+  CONSTANTS:
+    BEGIN OF co_extensibility_mode,
+      "! <p class="shorttext">Fully Supported Including Custom Analytical Queries</p>
+      "! Fully supported including custom analytical queries
+      fully_supported              TYPE ty_extensibility_mode VALUE 0,
+      "! <p class="shorttext">Supported Only for Predefined Analytical Queries</p>
+      "! Supported only for predefined analytical queries
+      supported_predefined_queries TYPE ty_extensibility_mode VALUE 1,
+      "! <p class="shorttext">Not Supported</p>
+      "! Not supported
+      not_supported                TYPE ty_extensibility_mode VALUE 2,
+    END OF co_extensibility_mode.
+
+
+  TYPES:
+    "! <p class="shorttext">Review Booklet Definition</p>
+    "! Review booklet definition
+    BEGIN OF ty_booklet_definition,
+      "! <p class="shorttext">Application ID</p>
+      "! Application ID
+      "! $required
+      application        TYPE ty_app_id,
+      "! <p class="shorttext">Source</p>
+      "! Source
+      "! $required
+      source             TYPE ty_source,
+      "! <p class="shorttext">Status</p>
+      "! Status
+      "! $required
+      status             TYPE ty_status,
+      "! <p class="shorttext">Consistency Status</p>
+      "! Consistency status
+      "! $required
+      consistency_status TYPE ty_consistency_status,
+      "! <p class="shorttext">Extensibility Mode</p>
+      "! Extensibility mode
+      "! $required
+      extensibility_mode TYPE ty_extensibility_mode,
+    END OF ty_booklet_definition.
+
+  TYPES:
+    "! <p class="shorttext">InA Service</p>
+    "! InA Service
+    BEGIN OF ty_ina1_service,
```

Comment:
I am wondering if some of these fields id , type are redundant, in a sense that they contain information that belongs to the InA Service, and not here. For example, if you change the type in the InAService, you will have obsolete information in your object. A symptom of this is that these fields are usually read only, without having had any input from the user. Is this the case?

Reply:
Hi Guilherme,
the ina1_service_id field will be the technical DB key for our persistency table, the value of it will be generated via our RAP implementations, so is read-only. ina1_service field will require user input to specify the desired InA service to be used, so is editable field. service_type is an internal concept => value can be either predefined InA service, or custom InA service. In the ADT editor case the value will be always with type 'predefined InA service'.

similarly, we also have some read-only fields from the ty_booklet_definition, would need clarifications how to be defined properly in the AFF.
would be great to have a short call to clarify these open points. Thanks!
BR, Ray

Reply:
Please don't set conversations to resolved, if questions have not been clarified, yet. Since this question seems to be still open. I reopen it.

Reply:
HI @schneidermic0 , will do for the future. This specific conversation was clarifed in the 1st sync meeting with Guilherme and therefore was closed. We have follow up conversation with similar topic, see below.

Reply:
Sorry, I didn't know you have clarified this in a call